### PR TITLE
ensure pointer's to arrays exist

### PR DESCRIPTION
**Pull request by bueller**

### CODE REVIEW

File: root_config.go [hunk 1]
```diff
@@ -11,10 +11,10 @@ type RootConfig struct {
 	State                  State            `yaml:"state" json:"state"`
 	Backup                 Backup           `yaml:"backup" json:"backup"`
 	Monitors               Monitors         `yaml:"monitors" json:"monitors"`
-	Components             []*Component     `yaml:"components" json:"components" validate:"dive"`
-	ConfigCommands         []*ConfigCommand `yaml:"cmds" json:"cmds"`
-	ConfigGroups           []*ConfigGroup   `yaml:"config" json:"config" validate:"dive"`
-	AdminCommands          []*AdminCommand  `yaml:"admin_commands" json:"admin_commands" validate:"dive"`
+	Components             []*Component     `yaml:"components" json:"components" validate:"dive,exists"`
+	ConfigCommands         []*ConfigCommand `yaml:"cmds" json:"cmds" validate:"exists"`
```
Comment (emosbaugh) [inline root_config.go]: should this have a dive?

**Comment by emosbaugh** [comment]
should this have exists but no dive?

``` go
Items                 []*ConfigItem          `yaml:"items" json:"items"
```

**Comment by emosbaugh** [comment]
couple questions

**Comment by bueller** [comment]
@emosbaugh going to leave the `exists` out of ConfigItem array validation until we update struct and add a dive

**Comment by bueller** [comment]
@emosbaugh please review additions

**Comment by emosbaugh** [comment]
lgtm

**Pull request merged**